File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        if len(strs) == 1:
            return [strs]
        
        hash_map = {}
        for str in strs:
            str_arr = [*str]
            str_arr.sort()
            new_str = ''.join(str_arr)
            if not new_str in hash_map:
                hash_map[new_str] = [str]           
            else:
                hash_map[new_str].append(str)
        
        return hash_map.values()

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    if (strs.length === 1) return [strs];
    
    let hash = {};
    let results = [];
    
    for (let str of strs) {
        const sorted = [...str].sort().join('');
        if (!(sorted in hash)) {
            hash[sorted] = [str];
        } else {
            hash[sorted].push(str);
        }
    }
    
    for (let key in hash) {
        results.push(hash[key]);
    }
        
    return results;
};